keps/sig-node/24-apparmor/README.md

```diff
@@ -0,0 +1,593 @@
+
+
+#### Pod Update
+
+The AppArmor fields on a pod are immutable, which also applies to the
+[annotation](https://github.com/kubernetes/kubernetes/blob/b46612a74224b0871a97dae819f5fb3a1763d0b9/pkg/apis/core/validation/validation.go#L177-L182).
+
+When an [Ephemeral Container](/keps/sig-node/277-ephemeral-containers/README.md) is added, it
+will follow the same rules for using or overriding the pod's AppArmor profile.
+Ephemeral container's will never sync with an AppArmor annotation.
+
+#### PodTemplates
+
+PodTemplates (e.g. ReplaceSets, Deployments, StatefulSets, etc.) will be
+ignored. The field/annotation resolution will happen on template instantiation.
+
+#### Warnings
+
+To raise awareness of existing controllers using the AppArmor annotations that need to be migrated,
+a warning will be emitted when annotations are set on pod creation, or pod template (including
+embedded pod templates) create & update.
+
+#### Runtime Profiles
+
+The API Server will continue to reject annotations with runtime profiles
+different than `runtime/default`, to maintain the existing behavior.
+
+Violations would lead to the error message:
+
+```
+Invalid value: "runtime/profile-name": must be a valid AppArmor profile
+```
+
+#### Upgrade / Downgrade Strategy
+
+Nodes do not currently support in-place upgrades, so pods will be recreated on
+node upgrade and downgrade. No special handling or consideration is needed to
+support this.
+
+On the API server side, we've already taken version skew in HA clusters into
+account. The same precautions make upgrade & downgrade handling a non-issue.
+
+Since
+[we support](https://kubernetes.io/docs/setup/release/version-skew-policy/) up
+to 2 minor releases of version skew between the master and node, annotations
+must continue to be supported and backfilled for at least 2 versions passed the
+initial implementation. Specifically, fields will no longer be copied to annotations for older kubelet
+versions. However, annotations submitted to the API server will continue to be copied to fields at the
+kubelet indefinitely, as was done with Seccomp.
+
+##### Kubelet Backwards compatibility
+
+Since we don't support running newer Kubelets than API server, new Kubelets only need to handle
+AppArmor fields. All the version skew resolution happens within the API server.
+
+### Graduation Criteria
+
+_This section is excluded, as it is the subject of the entire proposal._
+
+## Production Readiness Review Questionnaire
+
+### Feature enablement and rollback
+
+###### How can this feature be enabled / disabled in a live cluster?
+
+AppArmor is controlled by the `AppArmor` feature gate, but no new feature gate is added for the changes outlined in this proposal.
+
+- [X] Feature gate
+  - Feature gate name: `AppArmor`
+  - Components depending on the feature gate:
+      - kube-apiserver
+      - kubelet
+
+###### Does enabling the feature change any default behavior?
+
+No - AppArmor has been enabled by default since Kubernetes v1.4.
```

Comment:
```suggestion
No; you need to set an AppArmor profile on a Pod for the change to actually affect container / node behavior.
The `AppArmor` feature gate has been enabled by default since Kubernetes v1.4.
```